### PR TITLE
Proxy+SSL feature

### PR DESCRIPTION
**Pull request by jordipainan**
Added proxy and SSL with Let's encrypt

### CODE REVIEW

File: cmd/gateway/gateway.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ func main() {
 
 	dvoteHost := flag.String("dvoteHost", "0.0.0.0", "dvote API host")
 	dvotePort := flag.Int("dvotePort", 9090, "dvote API port")
-	dvoteRoute := flag.String("dvoteRoute", "/dvote", "dvote API route")
+	dvoteRoute := flag.String("dvoteRoute", "/dvote-ws", "dvote API route")
```
Comment (p4u) [inline cmd/gateway/gateway.go]: Please, use /dvote for legacy purposes

File: cmd/gateway/gateway.go [hunk 2]
```diff
@@ -42,6 +42,9 @@ func main() {
 	ipfsDaemon := flag.String("ipfsDaemon", "ipfs", "ipfs daemon path")
 	ipfsNoInit := flag.Bool("ipfsNoInit", false, "do not start ipfs daemon (if already started)")
 
+	sslDomain := flag.String("sslDomain", "", "ssl secure domain")
+	sslDirCert := flag.String("sslDircert", "./", "path where the ssl files will be stored")
```
Comment (p4u) [inline cmd/gateway/gateway.go]: Naming should be sslDirCert

**Comment by jordipainan** [comment]
closes #28 #29

**Comment by p4u** [comment]
There are conflicts. Please, fix them and push again :)

**Pull request merged**